GameOfLife.java

```diff
@@ -58,21 +60,44 @@ private static void play(String fileName) {
 	// and the rightmost columns. Thus the actual board is surrounded by a "frame" of zeros. You can think
 	// of this frame as representing the infinite number of dead cells that exist in every direction.
 	// This function assumes that the input file contains valid data, and does no input testing.
-	private static int[][] read(String fileName) {
+	public static int[][] read(String fileName) {
 		In in = new In(fileName); // Constructs an In object for reading the input file
 		int rows = Integer.parseInt(in.readLine());
 		int cols = Integer.parseInt(in.readLine());
 		int[][] board = new int[rows + 2][cols + 2];
 		//// Replace the following statement with your code.
-		return null;
+		String ln = "";
+
+		for (int i=1; i<=rows; i++){
+			ln = in.readLine(); 
+			if (ln != "") {
+				for (int j=1; j<=ln.length(); j++){
+					if (ln.charAt(j-1) == 'x'){
+						board[i][j] = 1; 
+					} 
+				}
+			}
```

Comment:
if ln == "" than it length is 0 so no iteration will be run through this loop - that means this if is not required.
```suggestion
				for (int j=1; j<=ln.length(); j++){
					if (ln.charAt(j-1) == 'x'){
						board[i][j] = 1; 
					} 
				}
```